### PR TITLE
feat: add column search feature

### LINKED ISSUE #36: [Task 13] Need to enable individual column searching
Check out the old react-datatable repo: https://bsol-oss.github.io/react-datatable/?path=/story/data-table--with-cell-height-width-variations
image
These are the search bar that will allow filtering just on an individual column.

Before start pls read again carefully the background I send over on Slack, and make sure you tried and understand how to work with the API (by checking out existing code how to form the querystring properly)

Again link to the old datatable repo https://github.com/bsol-oss/react-datatable

The search bar should mimic what the global search bar is doing:
1. let the user key in whatever
2. When search icon is clicked or pressed enter, trigger API call for the search

Any questions pls ask me in Slack.

### PR DESCRIPTION
**Pull request by Risk-taker0501**
closes #36 #37 

I finished task : individual column search like this:
```
<BodyWrapper columns={columns}>
          <ColumnSearchWrapper />
 </BodyWrapper>
```

### CODE REVIEW

**Comment by screw123** [comment]
Hi @Risk-taker0501 this is not what I'm looking for.  I need more flexibility.

Consider not all columns has the same way of filtering
1. name, description, are search bar
2. is_active is a dropdown
3. imagine we have a date field in future, and it should be a range / datepicker
4. etc
5. we can also skip it for some columns (mean taking away the column filtering of some columns)

your way of doing it, every column has the same.

My suggested way:
columns.tsx
```
import SearchBar
const columns = [
  { accessor: name, filter: SearchBar, ...}
  { accessor: is_active, filter: Dropdown}
]
```

in BodyWrapper, you adding somthing this:
```
{header.column.columnDef.filter ? (header.column.columnDef.filter) : <></>}
```

In short, we import components in the column definition.  And then we render the component in BodyWrapper if it's provided.  Otherwise we render nothing.

This method I suggested, is aligned with old repo.  You can check it out.
